concordium-consensus/src/Concordium/GlobalState/Basic/BlockState.hs

```diff
@@ -2076,19 +2075,16 @@ genesisState gd = case protocolVersion @pv of
                   _blockUpdates = initialUpdates genesisUpdateKeys genesisChainParameters
                   _blockReleaseSchedule = Map.empty
 
-        mkGenesisStateRegenesis migration GenesisData.RegenesisData{..} = do
-            case runGet (getBlockState migration) genesisNewState of
-                Left err -> Left $ "Could not deserialize genesis state: " ++ err
-                Right bs
-                    | hashShouldMatch && hbs ^. blockStateHash /= genesisStateHash -> Left "Could not deserialize genesis state: state hash is incorrect"
-                    | epochLength (bs ^. blockBirkParameters . birkSeedState) /= GenesisData.genesisEpochLength genesisCore -> Left "Could not deserialize genesis state: epoch length mismatch"
-                    | otherwise -> Right $!! (bs, genesisTT)
-                    where
-                        hbs = hashBlockState bs
-                        hashShouldMatch = case migration of
-                            StateMigrationParametersTrivial{} -> True
-                            StateMigrationParametersP3ToP4{} -> False
-                        genesisTT = getInitialTransactionTable hbs
+migrateBlockState ::
+    forall oldpv pv.
+    (IsProtocolVersion oldpv, IsProtocolVersion pv) =>
+    StateMigrationParameters oldpv pv ->
+    BlockState oldpv ->
+    Either String (BlockState pv)
+migrateBlockState migration oldbs = do
+    case runGet (getBlockState migration) (runPut $ putBlockState oldbs) of
```

Comment:
I don't think that will be helpful. This should be strict already since runGet ought to be.

But in any case this is for the basic state which does not matter and has a host of space leaks as it is.